### PR TITLE
Tech debt changes using List.of & Set.of

### PR DESCRIPTION
**Pull request by vladkovalskyi**
- [x] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/ParseRequestServiceTest.java [hunk 1]
```diff
@@ -179,10 +179,10 @@ void validateUserProfiles_throwsBadRequest_tooManyPrimaryTest() {
 
     @Test
     void validateUserProfiles_throwsResourceNotFound_noProfilesTest() {
-        List<CaseWorkerProfile> caseWorkerProfiles = new ArrayList<>();
+        List<CaseWorkerProfile> caseWorkerProfiles = List.of();
```
Comment (sankar-padakula) [inline src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/ParseRequestServiceTest.java:182]: use Collections.emptyList or new ArrayList here.
List.of is not required

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/ParseRequestServiceTest.java [hunk 2]
```diff
@@ -179,10 +179,10 @@ void validateUserProfiles_throwsBadRequest_tooManyPrimaryTest() {
...
         UserRequest userRequest = TestDataBuilder.buildUserRequest();
         AtomicInteger integer = new AtomicInteger();
-        Set<CaseWorkerProfile> invalidCaseWorkerProfiles = new HashSet<>();
+        Set<CaseWorkerProfile> invalidCaseWorkerProfiles = Set.of();
```
Comment (sankar-padakula) [inline src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/ParseRequestServiceTest.java:185]: use Collections.emptySet or new HashSet here.
Set.of is not required

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/ParseRequestServiceTest.java [hunk 3]
```diff
@@ -191,7 +191,7 @@ void validateUserProfiles_throwsResourceNotFound_noProfilesTest() {
 
     @Test
     void validateUserProfiles_throwsResourceNotFound_someProfilesNotFoundTest() {
-        List<CaseWorkerProfile> caseWorkerProfiles = new ArrayList<>();
+        List<CaseWorkerProfile> caseWorkerProfiles = List.of();
```
Comment (sankar-padakula) [inline src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/ParseRequestServiceTest.java:194]: use Collections.emptyList or new ArrayList here.
List.of is not required

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/RetrieveDataServiceTest.java [hunk 4]
```diff
@@ -119,7 +119,7 @@ void shouldReturnCaseWorkerProfile() {
 
     @Test
     void shouldReturnZeroCaseWorkerProfile() {
-        List<CaseWorkerProfile> caseWorkerProfiles = new ArrayList<>();
+        List<CaseWorkerProfile> caseWorkerProfiles = List.of();
```
Comment (sankar-padakula) [inline src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/RetrieveDataServiceTest.java:122]: use Collections.emptyList or new ArrayList here.
List.of is not required

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/RetrieveDataServiceTest.java [hunk 5]
```diff
@@ -168,7 +168,7 @@ void shouldReturnJudicialProfile() {
 
     @Test
     void shouldReturnZeroJudicialProfile() {
-        List<JudicialProfile> judicialProfiles = new ArrayList<>();
+        List<JudicialProfile> judicialProfiles = List.of();
```
Comment (sankar-padakula) [inline src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/RetrieveDataServiceTest.java:171]: use Collections.emptyList or new ArrayList here.
List.of is not required

**Review by aashishaggarwal123 — APPROVED** [review]

**Review by sankar-padakula — APPROVED** [review]

**Pull request merged**